### PR TITLE
3.0: Adapt Scheduling and Shared storage schema to new design

### PR DESCRIPTION
**Pull request by chenwany**
* Update SharedStorage sections by changing the generic Settings to ${StorageType}Settings (e.g. FsxLustreSettings, EbsSettings)
* Update Scheduling section
 * Configure the scheduler type as a value of the Scheduler parameter
   * Have specific ${Scheduler}Settings sections
   * Rename ComputeType to CapacityType in Queue section
   * Group compute specific settings under ComputeSettings section

### CODE REVIEW

File: cli/src/pcluster/templates/cdk_builder_utils.py [hunk 1]
```diff
@@ -49,10 +49,20 @@ def get_block_device_mappings(node_config: Union[HeadNode, BaseQueue], os: str):
         )
 
     # Root volume
-    if node_config.local_storage and node_config.local_storage.root_volume:
-        root_volume = copy.deepcopy(node_config.storage.root_volume)
-    else:
-        root_volume = Ebs()
+    if hasattr(node_config, "compute_settings"):  # Queue
```
Comment (hanwen-pcluste) [inline cli/src/pcluster/templates/cdk_builder_utils.py:52]: `if isinstance(node_config, BaseQueue):`

File: cli/src/pcluster/templates/cdk_builder_utils.py [hunk 2]
```diff
@@ -49,10 +49,20 @@ def get_block_device_mappings(node_config: Union[HeadNode, BaseQueue], os: str):
...
+        if (
+            node_config.compute_settings
+            and node_config.compute_settings.local_storage
+            and node_config.compute_settings.local_storage.root_volume
+        ):
+            root_volume = copy.deepcopy(node_config.compute_settings.local_storage.root_volume)
+        else:
+            root_volume = Ebs()
+    elif hasattr(node_config, "local_storage"):  # HeadNode
```
Comment (hanwen-pcluste) [inline cli/src/pcluster/templates/cdk_builder_utils.py:61]: `if isinstance(node_config, HeadNode):`

**Review by hanwen-pcluste — COMMENTED** [review]
Great work!

File: cli/src/pcluster/templates/cdk_builder_utils.py (hunk 1 above)
Comment (chenwany) [reply cli/src/pcluster/templates/cdk_builder_utils.py]: Changed

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**